### DISCUSSION TITLE
iOptron mount has problem communicating with telescope plugin

### DISCUSSION THREAD

**Discussion by muggsie** [Stellarium/stellarium#2506] [General]
My CEM70 mount is having a communication problem with Stellarium.  When I send it a slew command from Stellarium, the mount often takes a very long time to start slewing, and sometimes it does not slew at all.   I'm using Stellarium version 0.21.3 with Windows 10 Pro on a NUC that sits on the OTA and is directly connected to the mount by USB.   I am using the ASCOM option in the Stellarium telescope plugin and the iOptron Commander driver.   The driver is up to date.  

The Stellarium plugin always says the mount is connected, and the mount icon appears on the Stellarium map in the correct location.  But when I issue a slew command from Stellarium, the mount often does not move at all, or it takes from 10 seconds to a minute to start the slew.  Only once in while does it move immediately.   If I disconnect the mount in the telescope plugin and then reconnect it, sometimes the mount starts to slew, but not always.   If I close Stellarium and restart it, Stellarium remembers the unexecuted slew command and when it recconnects the mount, the slew starts immediately every time.    In all cases the icon representing the mount is present on the Stellarium screen indicating that Stellarium and mount are connected.   

NINA and PHD2 communicate with the mount without any problems.  I can also send commands to the mount directly from iOptron Commander without any problem.   I have been using all of them together for a year and half in the same configuration on the NUC, but this problem appeared in Stellarium a couple months ago.    

This issue is not preventing me from imaging, but it is slowing down my initial nightly setup.  Sometimes I have to restart Stellarium several times to complete my startup workflow.

Any help would be greatly appreciated.

**Comment by BlackApron** [comment]
I'm having the exact same problem with the same mount and Stellarium... Any updates on the issue?

Thanks.

**Reply by gzotti** [reply]
Nope, sorry. We definitely need more contributing programmer-observers who are willing to invest a few hours or nights of their time to track down the problems with these mounts.

**Comment by BlackApron** [comment]
Anything I can do to help?
________________________________

**Reply by gzotti** [reply]
Thanks for asking. I am sorry to say that I don't know this plugin too well as I have only a small mount that just worked whenever I tried. If you can compile the program, try to understand the source files of the TelescopeControl plugin, and start debugging.

**Comment by Gaw444** [comment]
Hi, I hope I'm not too late.
I have had the same problem with my CEM60EC when performing several updates a long time ago. After using the system for a long time I missed an important step in the configuration.

The issue is resolved by installing StellariumScope. In it, the mount is configured by choosing the correct driver (the one from iOptron) and then sending said configuration to Stellarium from StellariumScope.

All problems will be fixed. In any case, check that in Stellarium Telescope Control is activated at startup.

This works on Win10 and 11 (I assume Win7 too) with the latest drivers from iOptron, Ascom and Stellarium (Oct 23/2022).

If you check the configuration of the telescope in Stellarium you will see that it uses the communication with Ascom not directly but through software (the iOptron driver).

Here StellariumScope:  https://welshdragoncomputing.ca/x/index.php/home/stellariumscope/download-stellariumscope

Good luck, good heavens

**Reply by gzotti** [reply]
We actually want to get away from this former necessity. Something around ASCOM still seems not finished in connection with iOptron. But we need a developer with such telescope or a telescope owner who at least has enough programming experience to dive into this.

**Comment by BlackApron** [comment]
I tried this and it broke Stellarium 1.0... It doesn't even load now.

I've tried uninstalling and reinstalling but nothing... don't know what it did.

🙁

Any logs I can share to troubleshoot?
________________________________

**Comment by BlackApron** [comment]
This is what is appearing in the log...

*** Problematic Fragment shader source code ***
#version 110
#ifdef GL_KHR_blend_equation_advanced
#extension GL_ARB_fragment_coord_conventions : enable
#extension GL_KHR_blend_equation_advanced : enable
#endif
#define lowp
#define mediump
#define highp
#line 1
varying highp vec2 uv;uniform sampler2DRect textureSampler;uniform bool swizzle;uniform highp float opacity;void main() {   highp vec4 tmpFragColor = texture2DRect(textureSampler,uv);   tmpFragColor.a *= opacity;   gl_FragColor = swizzle ? tmpFragColor.bgra : tmpFragColor;}
***

________________________________

**Reply by gzotti** [reply]
That's a problem with your graphics system. Try the alternative build of 1.0, V0.22.3.

**Comment by BlackApron** [comment]
Thanks... I will... but just to be clear,  1.0 worked perfectly fine in the same computer before trying this.
________________________________

**Reply by gzotti** [reply]
Wait. You installed stock 1.0 (1.22.3) and it worked, then you configured the telescope, and suddenly your graphics system fails? That's magic. 
Or did you build current master with some added options?

**Comment by BlackApron** [comment]
I've had stock 1.22.3 installed for a couple of weeks working perfectly fine.

Then, I tried this morning the Stellarium Telescope suggestion and 1.22.3 stopped loading.

I've installed 0.22.3 and it is working.

I know is weird but that is what happened.

🙁
________________________________

**Reply by gzotti** [reply]
"Stellarium Telescope suggestion": Do you mean StellariumScope?

**Comment by BlackApron** [comment]
Yes... the one about configuring the scope in StellariumScope and sending it to Stellarium...
________________________________

**Reply by gzotti** [reply]
I cannot help on this one. Maybe @Gaw444 knows more.

**Comment by Gaw444** [comment]
In my case I had problems updating to Stellarium 1.0 and communicating with the mount. I went back to the previous version (0.22.x), used StellariumScope again and everything works correctly and perfect connection (PHD2, APT, Stellarium, weather sensors, etc.), all with Ascom.

At the time of starting the session, I already have everything configured with StellariumScope. I run StellariumScope from which I activate the "connect" box (it launches the iOptrom Commander that controls the mount. I minimize it since I do the Gotos from Stellarium or APT), then from StellariumScope hit "start Stellarium" and you can already see where the mount is pointing on it. I then start PHD2 as usual and then APT (or Sharpcap, Nina, etc.).

I understand that the connection problem using version 1.0 of Stellarium, in my case, is related to the graphics management (I use two monitors) although I cannot confirm it since I went back to the previous version and did not investigate this issue.

System I use: Industrial MiniPC on the pier (I7, 8 USB 3.1 ports; 4 RS232 ports; 2 Wi-Fi, 2 ethernet; 2 monitors; Win10 64 updated...).
iOptron CEM60EC mount with latest Ascom drivers (6.5x) and iOptron Commander 5.9.0.2. The mount is up to date with the latest firmware (Main Control, HandControl, RA and Dec).

I hope that by testing with version 0.22.x of Stellarium and configuring Ascom communication with StellariumScope everything works fine.

In my opinion it is preferable to update one element at a time, in order to be able to correct the error if necessary.

Best regards
